src/net.h

```diff
@@ -485,24 +485,19 @@ class V2Transport final : public Transport
         /** Garbage and garbage terminator.
          *
          * Whenever a byte is received, the last 16 bytes are compared with the expected garbage
-         * terminator. When that happens, the state becomes GARBAUTH. If no matching terminator is
+         * terminator. When that happens, the state becomes VERSION. If no matching terminator is
          * received in 4111 bytes (4095 for the maximum garbage length, and 16 bytes for the
          * terminator), the connection aborts. */
         GARB_GARBTERM,
 
-        /** Garbage authentication packet.
-         *
-         * A packet is received, and decrypted/verified with AAD set to the garbage received during
-         * the GARB_GARBTERM state. If that succeeds, the state becomes VERSION. If it fails the
-         * connection aborts. */
-        GARBAUTH,
-
         /** Version packet.
          *
-         * A packet is received, and decrypted/verified. If that succeeds, the state becomes APP,
-         * and the decrypted contents is interpreted as version negotiation (currently, that means
-         * ignoring it, but it can be used for negotiating future extensions). If it fails, the
-         * connection aborts. */
+         * A packet is received, and decrypted/verified. The first received packet in this state
+         * (whether it's a decoy or not) is expected to authenticate the garbage received during
+         * the GARB_GARBTERM state as AAD. If decryption/verification (of the first non-decoy
+         * decoy) succeeds, the state becomes APP, and the decrypted contents is interpreted as
+         * version negotiation (currently, that means ignoring it, but it can be used for
```

Comment:
This description is confusing.
Currently, we will switch to APP no matter whether the authenticating package was decoy or version.

In future, `the state becomes APP, and the decrypted contents is interpreted as version negotiation`might not be the case, if the authenticating package was decoy.

At that point we might as well change this comment, but why having something certainly confusing in the first place?

Reply:
> In future, `the state becomes APP, and the decrypted contents is interpreted as version negotiation` might not be the case, if the authenticating package was decoy.

`the state becomes APP, and the decrypted contents is interpreted as version negotiation` *is* always the case. The first non-decoy packet received in VERSION state is the version packet, and that means that we interpret the contents as version info and switch to APP.

> At that point we might as well change this comment, but why having something certainly confusing in the first place?

I'm not sure what you're suggesting. Can you elaborate? (Maybe it's based on a misunderstanding that has been resolved now?)

Reply:
I think my issue was resolved with the typo fix right above.